interface.js

```diff
@@ -0,0 +1,436 @@
+console.log('interface.js loaded');
+const Store = require('electron-store');
+const store = new Store();
+
+/* ========================================================================== */
+/* Provider-Specific Handlers                                                 */
+/* ========================================================================== */
+
+/* Shared Handlers ---------------------------------------------------------- */
+
+class Provider {
+	static setupCustomPasteBehavior() {
+		this.webview.addEventListener(
+			'dom-ready',
+			function () {
+				this.webview.executeJavaScript(`
+			document.addEventListener('paste', (event) => {
+				event.preventDefault();
+				let text = event.clipboardData.getData('text');
+				let activeElement = document.activeElement;
+				let start = activeElement.selectionStart;
+				let end = activeElement.selectionEnd;
+				activeElement.value = activeElement.value.slice(0, start) + text + activeElement.value.slice(end);
+				activeElement.selectionStart = activeElement.selectionEnd = start + text.length;
```

Comment:
i somewhat worry about overriding paste like this, lets make sure to check it works in the main chat apps